nixos/modules/virtualisation/google-compute-image.nix

```diff
@@ -44,10 +44,30 @@ in
         GZIP compression level of the resulting disk image (1-9).
       '';
     };
+
+    virtualisation.googleComputeImage.efi = mkOption {
+      type = types.bool;
+      default = cfg.nvme;
+      description = lib.mdDoc ''
+        Whether to use EFI booting.
+      '';
+    };
+
+    virtualisation.googleComputeImage.nvme = mkOption {
+      type = types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Whether to support NVMe devices.
+      '';
+    };
   };
 
   #### implementation
   config = {
+    boot.initrd.availableKernelModules = mkIf cfg.nvme [ "nvme" ];
```

Comment:
nvme is already included by default, see nixos/modules/system/boot/kernel.nix and https://search.nixos.org/options?channel=unstable&show=boot.initrd.includeDefaultModules&from=0&size=50&sort=relevance&type=packages&query=includeDefault

We can add it in case the user disables includeDefaultModules, but then please do that unconditionally

Reply:
Updated